src/dynamics/video_player/player/player.js

```diff
@@ -3084,37 +3085,36 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
                  * @private
                  */
                 __controlAdRolls: function() {
+                    const shouldPlayMidrolls = this.get("duration") > this.get("max_shortform_video_duration");
+                    const hasMidRolls = this.get("midrollads").length > 0;
                     // If we have mid-rolls, then prepare mid-Rolls
-                    if (
-                        this.get("midrollads").length > 0 && this.get("duration") > 0.0 && !this._adsRollPositionsCollection
-                    ) {
+                    if (shouldPlayMidrolls && hasMidRolls && this.get("duration") > 0.0 && !this._adsRollPositionsCollection) {
                         this._adsRollPositionsCollection = this.auto_destroy(new Collection()); // our adsCollections
-                        if (this.get("midrollads").length > 0) {
-                            let _current = null;
-                            Objs.iter(this.get("midrollads"), function(roll, index) {
```

Comment:
Just get rid of `Obj.iter` and use the native `forEach`, an line 3092 was unnecessary since it was already validate on line 3091 `hasMidRolls`